### ISSUE TITLE
Some times we may want to provide test set in addition to training and validation sets

### ISSUE THREAD

**Issue by wschin** [dotnet/machinelearning#1572]
Imagining that one day, a user prepares a training set and a validation, and she wants to train a `FastTree` (gradient boosting decision tree) with early stopping. Then, she probably also would like to know how good our early stopping is by calculating a score based on some predictions. If this is the case, she may need another data set (let's call it test set) because it's not fair to use validation set again because the training iteration number is determined by validation set via early stopping.
